[DOCS-1571] BYOB on CoreWeave storage

mdlinville opened the pull request:
[DOCS-1571] BYOB on CoreWeave storage
- Available for Dedicated Cloud and Self-Hosted instances and teams
- Not yet available for Multi-tenant Cloud teams

- Ready for technical review 
- Ready for **early** peer review
- Link checker [passes](https://github.com/wandb/docs/actions/runs/15666125208/job/44130455884)

### Previews
Major changes:
- https://docs-1571-mvp.docodile.pages.dev/guides/hosting/data-security/secure-storage-connector/

Minor changes:
- https://docs-1571-mvp.docodile.pages.dev/guides/hosting/data-security/presigned-urls/
- https://docs-1571-mvp.docodile.pages.dev/guides/hosting/
- https://docs-1571-mvp.docodile.pages.dev/guides/hosting/hosting-options/saas_cloud/
- https://docs-1571-mvp.docodile.pages.dev/guides/hosting/hosting-options/dedicated_cloud/
- https://docs-1571-mvp.docodile.pages.dev/guides/hosting/hosting-options/self-managed/
- https://docs-1571-mvp.docodile.pages.dev/guides/hosting/self-managed/ref-arch/
- https://docs-1571-mvp.docodile.pages.dev/guides/hosting/monitoring-usage/audit-logging/
- https://docs-1571-mvp.docodile.pages.dev/ref/python/data-types/table/ (TODO fix the bad link in the API upstream)

Out of scope, will be handled in followup:
https://wandb.atlassian.net/browse/DOCS-1608 Update or add `cw://` examples where we have `s3://` examples, in both BYOB and run / artifact docs.

[DOCS-1571]: https://wandb.atlassian.net/browse/DOCS-1571?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

mdlinville committed:
[DOCS-1571] Update BYOB docs for FullyConnected (WIP)

mdlinville committed:
[DOCS-1571] Update BYOB docs for FullyConnected (WIP)

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -7,118 +7,109 @@ title: Bring your own bucket (BYOB)
 
-## Configuration options
-There are two scopes you can configure your storage bucket to: at the *Instance level* or at a *Team level*. 
+### Bucket scopes
```
Recast this section a bit

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -7,118 +7,109 @@ title: Bring your own bucket (BYOB)
-{{% /alert %}}
+### Availability matrix
+W&B can connect to the following types of storage:
+- [CoreWeave AI Object Storage](https://docs.coreweave.com/docs/products/storage/object-storage) is a high-performance, S3-compatible object storage service optimized for AI workloads.
```
We didn't previously list all the options in one place. I also reworked the availability matrix table a bit.

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -7,118 +7,109 @@ title: Bring your own bucket (BYOB)
-</details>
+<sup><a id="footnote_1">1</a>.</sup> CoreWeave AI Object Storage is currently supported only for team level BYOB. Instance level support for Dedicated Cloud and Self-Hosted is coming soon.
 
+## Set up BYOB
```
Restructured the headings here to show the sequence better.

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -7,118 +7,109 @@ title: Bring your own bucket (BYOB)
+After [verifying availability]({{< relref "#availability-matrix" >}}), you are ready to provision your storage bucket, including its access policy and CORS. Select a tab to continue.
 
-Based on your use case, configure a storage bucket at the team or instance level. How a storage bucket is provisioned or configured is the same irrespective of the level it's configured at, except for the access mechanism in Azure. 
+{{< tabpane text=true >}}
```
storage-specific tabs

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -297,58 +289,93 @@ For more details, see the [AWS self-managed hosting guide]({{< relref "/guides/h
 {{< /tabpane >}}
 
-## Configure BYOB in W&B
+### 2. Determine the storage address  {#determine-the-storage-address}
```
Rewrote the previous section https://github.com/wandb/docs/pull/1391/files#diff-19bb6d6657e85724c9ba35b5c955d80243bde09c323e74c9663951a1f0a92021L65 to have storage-specific tabs and give more details.

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -297,58 +289,93 @@ For more details, see the [AWS self-managed hosting guide]({{< relref "/guides/h
-{{< img src="/images/hosting/prod_setup_secure_storage.png" alt="" >}}
-
-An error or warning appears at the bottom of the page if there are issues accessing the bucket or the bucket has invalid settings.
+For Cloud-native storage buckets with an optional S3-compatible mode, use the Cloud-native protocol specifier when possible. For example, use `cw://` for a CoreWeave bucket, rather than `s3://`.
```
The intent here is to discourage using S3-compatibility mode when possible. In the previous content, I couldn't actually see anything different with same-cloud and cross-cloud syntax, apart from the S3 region requirement detail. If I missed it, let me know so I can explain it more clearly.

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -297,58 +289,93 @@ For more details, see the [AWS self-managed hosting guide]({{< relref "/guides/h
 {{% /alert %}}
 
-To make changes to the instance level bucket storage for your W&B Server:
+#### Instance level BYOB
```
New sections so we can link directly to them. Also it looks like you can configure instance level BYOB in the app, so I tabbed this content.

mdlinville committed:
Fixing up product names without going crazy

mdlinville committed:
Fixing up product names without going crazy

mdlinville committed:
Pre-signed URL updates

mdlinville committed:
Pre-signed URL updates

mdlinville committed:
Add troubleshooting accordion to CoreWeave setup instructions

mdlinville committed:
Add troubleshooting accordion to CoreWeave setup instructions

mdlinville committed:
Comment out unresolved TODOs for now

mdlinville committed:
Comment out unresolved TODOs for now

mdlinville committed:
Add redirects

mdlinville committed:
Add redirects

mdlinville committed:
Fix a link

mdlinville committed:
Fix a link

mdlinville committed:
Remove anchor-level redirects which don't work apparently

mdlinville committed:
Remove anchor-level redirects which don't work apparently

mdlinville committed:
Fix link better

mdlinville committed:
Fix link better

mdlinville committed:
Revert redirects

mdlinville committed:
Revert redirects

mdlinville committed:
Updates for CoreWeave bucket policioes, reorganization and copyedit

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1571-mvp

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1571-mvp' into DOCS-1571-mvp

trane293 commented on content/en/guides/hosting/data-security/presigned-urls.md:
```diff
@@ -34,26 +35,31 @@ W&B recommends adding users to only the teams that they are supposed to be a par
 
-In case of AWS, one can use [VPC or IP address based network restriction](https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-presigned-url.html#PresignedUrlUploadObject-LimitCapabilities). It ensures that your W&B specific buckets are accessed only from networks where your AI workloads are running, or from gateway IP addresses that map to your user machines if your users access artifacts ...
+In CoreWeave, AWS S3 or S3-compatible storage, you can restrict access using VPC or IP address based network restrictions to ensure that your W&B specific buckets are accessed only from networks where your AI workloads are running, or from gateway IP addresses that map to your user machines if your users access artifacts using the W&B UI. Refer to:
+- [CoreWeaVe documentation](https://docs.coreweave.com/docs/products/storage/object-storage/reference/bucket-policy#condition)
```
typo - "CoreWeaVe"

mdlinville replied:
Done

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1571-mvp

mdlinville committed:
Adjust BYOB docs for CoreWeave availability

- Not yet available at all for Multi-tenant Cloud
- Available for both instances and teams for Dedicated Cloud and Self-Hosted

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1571-mvp

ngrayluna commented on content/en/guides/hosting/data-security/presigned-urls.md:
```diff
@@ -34,26 +35,31 @@ W&B recommends adding users to only the teams that they are supposed to be a par
 W&B recommends restricting the networks that can use pre-signed URLs to access the blob storage, by using IAM policy based restrictions on the buckets. 
 
-In case of AWS, one can use [VPC or IP address based network restriction](https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-presigned-url.html#PresignedUrlUploadObject-LimitCapabilities). It ensures that your W&B specific buckets are accessed only from networks where your AI workloads are running, or from gateway IP addresses that map to your user machines if your users access artifacts ...
+In CoreWeave, AWS S3 or S3-compatible storage, you can restrict access using VPC or IP address based network restrictions to ensure that your W&B specific buckets are accessed only from networks where your AI workloads are running, or from gateway IP addresses that map to your user machines if your users access artifacts using the W&B UI. Refer to:
```
Nit: Long sentence. Consider breaking into two.

mdlinville replied:
Done

ngrayluna commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -7,118 +7,236 @@ title: Bring your own bucket (BYOB)
-{{% alert color="secondary" %}}
-Plan your storage bucket layout carefully. After you configure a storage bucket for W&B, migrating its data to another bucket is complex and requires the assistance of W&B. This applies to storage for Dedicated Cloud and Self-Managed, as well as team-level storage for SaaS Cloud. For questions, contact [support](mailto:support@wandb.com).
-{{% /alert %}}
+<sup><a id="footnote_1">1</a>.</sup> CoreWeave AI Object Storage is currently supported only for instance or team level BYOB on Dedicated Cloud and Self-Managed. Team level BYOB on Multi-tenant Cloud is coming soon.
```
Let's remove this? It dates the docs.

mdlinville replied:
I was asked to keep this in, hopefully the limitation will be lifted very soon.

ngrayluna commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -7,118 +7,236 @@ title: Bring your own bucket (BYOB)
+| W&B deployment type | Instance level                         | Team level | Additional information |
+|----------------------------|----------------------------------------|------------|------------------------|
+| Dedicated Cloud            | &check;<sup><a href="footnote_1">1</a></sup> | &check;          | Instance and team level BYOB are supported for CoreWeave AI Object Storage, AWS S3, GCP Storage, Microsoft Azure Blob Storage, and S3-compatible storage like [MinIO](https://github.com/minio/minio) hosted in your cloud or on-prem infrastructure. |
+| Multi-tenant Cloud                 | Not Applicable                         | &check;<sup><a href="footnote_1">1</a></sup> | Team level BYOB is supported for AWS S3, and GCP Storage. W&B fully manages the default and only storage bucket for Microsoft Azure. Team level BYOB for CoreWeave AI Object Storage is coming soon. |
```
Same with this one. consider removing (if not pressured by product to keep this).

mdlinville replied:
As with the other one, I need to leave this in for now. The plan was originally to deliver Multi-tenant first, and it is expected to be finished up very soon.

ngrayluna commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
````diff
@@ -7,118 +7,236 @@ title: Bring your own bucket (BYOB)
+    ```
 
-## Cloud storage in same cloud as W&B platform
+    </details>-->
````
There is a trailing "-->" showing up in the docs: 

<img width="799" alt="Screenshot 2025-06-27 at 1 19 42 PM" src="https://github.com/user-attachments/assets/b675d677-989a-4c04-9bde-84a5590d9a48" />

mdlinville replied:
Fixed

ngrayluna replied:
Not sure if my preview is caching, but I still see the extraneous "->". https://docs-1571-mvp.docodile.pages.dev/guides/hosting/data-security/secure-storage-connector/#set-up-byob

mdlinville replied:
Yeah you are right, hopefully fixed adfter this push!

ngrayluna commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -7,118 +7,236 @@ title: Bring your own bucket (BYOB)
-{{% /alert %}}
+<details>
+<summary>Troubleshoot CoreWeave storage connection errors</summary>
+<h4>Troubleshoot CoreWeave storage connection errors</h4>
```
Don't think we need this h4: 
<img width="814" alt="Screenshot 2025-06-27 at 1 23 09 PM" src="https://github.com/user-attachments/assets/a0d9b59c-8dfa-4458-89e5-29bed7279881" />

mdlinville replied:
Done (also moved the whole troubleshooting section to the end)

ngrayluna commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -357,5 +523,48 @@ To make changes to the instance level bucket storage for your W&B Server:
+- For Instance level BYOB, refer to [Instance level BYOB]({{< relref "#instance-level-byob" >}}) instead.
+{{% /alert %}}
+
+1. <!--If you are configuring CoreWeave storage for Multi-tenant Cloud, switch to the browser window where you previously began to create the new team to find the W&B organization ID previously. Otherwise, l-->Log in to W&B as a user with the `admin` role, click the icon at the top left to open the left navigation, then click **Create a team to collaborate**.
```
Seems like this comment modifies the desired emphasis on "Create a team to collaborate**
<img width="440" alt="Screenshot 2025-06-27 at 1 28 37 PM" src="https://github.com/user-attachments/assets/dc6f5f58-f86d-4acf-b0e6-2e57756e9621" />

mdlinville replied:
Done

ngrayluna commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -357,5 +523,48 @@ To make changes to the instance level bucket storage for your W&B Server:
+1. Provide a name for the team.
+1. Set **Storage Type** to **External storage**.
+1. Click **Bucket location**. Select an existing bucket or click **Add bucket** at the bottom to create a new bucket.
+    To add a new bucket:
```
Aww. Do we now have lettered ordered lists? Might make it more clear?

<img width="859" alt="Screenshot 2025-06-27 at 1 29 33 PM" src="https://github.com/user-attachments/assets/7e7d59d5-69d5-4e8d-8fa7-705320052784" />

mdlinville replied:
It turns out no, letters are not supported in Hugo! I reorganized this section a bit instead. PTAL.

ngrayluna commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -357,5 +523,48 @@ To make changes to the instance level bucket storage for your W&B Server:
+        - **Azure**: Set **Tenant ID** and **Managed Identity Client ID** fields to the appropriate values.
+        - **Multi-tenant Cloud**: Optionally invite members to the team. In **Invite team members**, specify a comma-separated list of email addresses. Otherwise, you can invite members to the team after it is created.
+
+          For **Dedicated Cloud** or **Self-Managed**, you can invite members to the team after it is created. 
```
Should this have a bullet point as well?

mdlinville replied:
No, it is part of the inviting members bullet point (now step 6). I verified it looks how I wanted it to. PTAL.

mdlinville committed:
Doc temporary limitations and mitigations for CoreWeave

- Contact W&B to configure CoreWeave for instance level BYOB
- Contact W&B before configuring CoreWeave for team level BYOB to validate things
- Move CoreWeave troubleshooting to new Troubleshooting section at the end, since it is at the W&B side

mdlinville committed:
Fix link

mdlinville committed:
Partially update Azure details for BYOB

trane293 commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
````diff
@@ -7,118 +7,209 @@ title: Bring your own bucket (BYOB)
+    - In `Principal` and `Condition`, each instance of `<wb-org-id>` with your W&B organization ID.
 
-## Cloud storage in same cloud as W&B platform
+    ```json
````
This is related to SaaS, this shouldn't go out on 7/1

mdlinville replied:
It's tricky to follow in the diff, but this is commented out. https://github.com/wandb/docs/pull/1391/files#diff-19bb6d6657e85724c9ba35b5c955d80243bde09c323e74c9663951a1f0a92021R147 Check the HTML preview linked in the PR description to see what the output looks like.

mdlinville committed:
Noah's feedback about nested sub-steps

mdlinville committed:
Anmol's feedback

estellazx commented on content/en/guides/hosting/data-security/presigned-urls.md:
```diff
@@ -24,36 +25,40 @@ A pre-signed URL expires after:
-W&B recommends restricting the networks that can use pre-signed URLs to access the blob storage, by using IAM policy based restrictions on the buckets. 
-
-In case of AWS, one can use [VPC or IP address based network restriction](https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-presigned-url.html#PresignedUrlUploadObject-LimitCapabilities). It ensures that your W&B specific buckets are accessed only from networks where your AI workloads are running, or from gateway IP addresses that map to your user machines if your users access artifacts ...
+In CoreWeave, AWS S3, or S3-compatible storage, you can define network restrictions at the VPC or IP address level. Refer to:
```
I'm not sure this works for coreweave, I don't think we should include coreweave here unless we know it works

mdlinville replied:
It looks like maybe not for VPC. I looked here:  https://docs.coreweave.com/docs/products/storage/object-storage/reference/bucket-policy#condition WDYT?

mdlinville replied:
Updated

ngrayluna reviewed (approved):
Minor nit (see comment). Approving so it doesn't block.

estellazx commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -7,118 +7,209 @@ title: Bring your own bucket (BYOB)
-Specify the path in a format specific to the locations of your W&B instance and storage bucket:
+    Replace:
+      - In `Resource`, each instance of `<cw_bucket>` with the CoreWeave bucket name.
+      - In `Principal`, each instance of `<cw_principal_and_role_arn>` with the CoreWeave principal and role ARN in the format `"arn:aws:iam:::user/<CW_ORG>:<CW_USER>"`. To apply the policy to all principals, replace the entire ARN with `*`. To apply the policy to all principals in your CoreWeave organization, replace `<CW_USER>` with `*`.
```
```suggestion
```
I'm confused what this is for? Do you mind showing me? I don't think this is relevant for CoreWeave server unless I'm missing something here

estellazx replied:
also its outdated as we decided on 
```
"Principal": {
     "CW": "arn:aws:iam::wandb:static/wandb-integration"
}
```

mdlinville replied:
this was probably confusing, so I have removed the previously commented multi-tenant stuff. The replacements in line 113-114 refer to:

```
"Principal": { "CW": "<cw_principal_and_role_arn>" },
```

and

```
          "Resource": [
            "arn:aws:s3:::<cw_bucket>",
            "arn:aws:s3:::<cw_bucket>/*"
```
respectively. This was the policy that Anmol shared with me for Multi-tenant, with the conditional removed. Is there a more up to date policy for me to adapt? What should the Resource look like? I've updated the principal.

estellazx commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
````diff
@@ -7,118 +7,209 @@ title: Bring your own bucket (BYOB)
-```text
-az://:<urlEncodedAccessKey>@<storageAccountName>/<containerName>
-```
+    ```json
````
this is also only for Saas, this entire section needs  to be commented out or deleted

mdlinville replied:
Now removed since the comments didn't work out, I'll use this PR's history to bring it back when we need it.

estellazx commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
````diff
@@ -297,58 +390,112 @@ For more details, see the [AWS self-managed hosting guide]({{< relref "/guides/h
+or
+```text
+s3://<accessKey>:<secretAccessKey>@<coreweaveEndpoint>/<bucketName>?region=<region>&tls=true
+```
````
```suggestion
```
lets remove this as we want `cw` to be the primary schema

mdlinville replied:
Done

estellazx commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -297,58 +390,112 @@ For more details, see the [AWS self-managed hosting guide]({{< relref "/guides/h
+- CoreWeave uses availability zones for AI Object Storage. In the bucket address, the `region` parameter is mandatory and must be set to the name of the CoreWeave availability zone that matches the CoreWeave bucket's location. Refer to [Regions and Availability Zones](https://docs.coreweave.com/docs/platform/regions) in the CoreWeave documentation. Click the **AI Object Storage** link for the reg ...
+- Replace `<coreweaveEndpoint>` with one of:
+  - `cwobject.com`: Primary HTTPS endpoint, TLS 1.3 required.
+  - `cwlota.com`: CoreWeave's [Local Object Transfer Accelerator (LOTA)](https://docs.coreweave.com/docs/products/storage/object-storage/concepts/lota) HTTP (not HTTPS) endpoint. LOTA is an intelligent proxy installed on every GPU Node in a CKS cluster to accelerate data transfer by providing an efficient local gateway to CoreWeave AI Object Storage on each node in the cluster for faster data tra ...
```
```suggestion
- legacy endpoints such as `object.lga1.coreweave.com`
- If there are other endpoints you would like to use please contact support @....
```

mdlinville replied:
Is it correct to remove LOTA? I understood that it was in scope for the initial release and I thought it was a CoreWeave-specific optimization. The legacy endpoint thing is news to me as I didn't even see it in the CoreWeave docs.

estellazx commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -297,58 +390,112 @@ For more details, see the [AWS self-managed hosting guide]({{< relref "/guides/h
+  - `cwobject.com`: Primary HTTPS endpoint, TLS 1.3 required.
+  - `cwlota.com`: CoreWeave's [Local Object Transfer Accelerator (LOTA)](https://docs.coreweave.com/docs/products/storage/object-storage/concepts/lota) HTTP (not HTTPS) endpoint. LOTA is an intelligent proxy installed on every GPU Node in a CKS cluster to accelerate data transfer by providing an efficient local gateway to CoreWeave AI Object Storage on each node in the cluster for faster data tra ...
+  See the [CoreWeave documentation](https://docs.coreweave.com/docs/products/storage/object-storage/how-to/get-started-caios#3-create-a-bucket) for details.
+- The `cw://` protocol specifier is preferred. However, CoreWeave buckets offer an optional S3-compatible mode when you use the `s3://` protocol specifier instead.
```
just remove the S3 part

mdlinville replied:
Done

estellazx commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -357,5 +504,85 @@ To make changes to the instance level bucket storage for your W&B Server:
+{{% /tab %}}
+{{% tab header="Environment variable" %}}
+
+To configure instance level BYOB, set the `GORILLA_SUPPORTED_FILE_STORES` environment variable to the bucket location, then restart W&B. 
```
This is wrong `GORILLA_SUPPORTED_FILE_STORES` is only used for server TEAM level byob

mdlinville replied:
Done

estellazx commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -297,58 +390,112 @@ For more details, see the [AWS self-managed hosting guide]({{< relref "/guides/h
 
-An error or warning appears at the bottom of the page if there are issues accessing the bucket or the bucket has invalid settings.
-{{% /tab %}}
+For **Self-Managed**, you can configure instance level BYOB using the W&B App or the `GORILLA_SUPPORTED_FILE_STORES` environment variable. Select a tab to continue.
```
TEAM level not instance level for `GORILLA_SUPPORTED_FILE_STORES` env variable

mdlinville replied:
Got it, and it is a prerequisite for configuring team level BYOB across clouds? Not an alternative way to configure team level BYOB.

mdlinville replied:
I believe this is done.

estellazx commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -357,5 +504,85 @@ To make changes to the instance level bucket storage for your W&B Server:
 1. Click **Save**
+
+{{% /tab %}}
+{{% tab header="Environment variable" %}}
```
I believe we can just remove this whole tab

mdlinville replied:
Done

estellazx commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -357,5 +504,85 @@ To make changes to the instance level bucket storage for your W&B Server:
+{{% /alert %}}
+
+#### Team level BYOB
+After you [determine the storage location](#determine-the-storage-location) for your bucket, you can use the W&B App to configure team level BYOB while creating a team.
```
```suggestion
After you [determine the storage location](#determine-the-storage-address) for your bucket, you can use the W&B App to configure team level BYOB while creating a team.
```

mdlinville committed:
Update content/en/guides/hosting/data-security/secure-storage-connector.md

Co-authored-by: Estella Xin <estellazx@gmail.com>

mdlinville committed:
Estella's feedback

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1571-mvp' into DOCS-1571-mvp

estellazx commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
````diff
@@ -297,58 +390,112 @@ For more details, see the [AWS self-managed hosting guide]({{< relref "/guides/h
+```text
+s3://<accessKey>:<secretAccessKey>@<coreweaveEndpoint>/<bucketName>?region=<region>&tls=true
+```
+- CoreWeave uses availability zones for AI Object Storage. In the bucket address, the `region` parameter is mandatory and must be set to the name of the CoreWeave availability zone that matches the CoreWeave bucket's location. Refer to [Regions and Availability Zones](https://docs.coreweave.com/docs/platform/regions) in the CoreWeave documentation. Click the **AI Object Storage** link for the reg ...
````
This is more of a nit as its fine to include the region but since coreweave buckets are globally unique, we don't actually need to specify the region. Might be easier to just remove this parameter and have 
`cw://<accessKey>:<secretAccessKey>@<coreweaveEndpoint>/<bucketName>?tls=true`
instead

mdlinville replied:
Done

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
````diff
@@ -297,58 +325,103 @@ For more details, see the [AWS self-managed hosting guide]({{< relref "/guides/h
+{{% tab header="CoreWeave" value="coreweave" %}}
+**Bucket format**:
+```text
+cw://<accessKey>:<secretAccessKey>@<coreweaveEndpoint>/<bucketName>?region=<region>&tls=true
````
```suggestion
cw://<accessKey>:<secretAccessKey>@cwobject.com/<bucketName>?tls=true
```

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -297,58 +325,103 @@ For more details, see the [AWS self-managed hosting guide]({{< relref "/guides/h
+- CoreWeave uses availability zones for AI Object Storage. In the bucket address, the `region` parameter is mandatory and must be set to the name of the CoreWeave availability zone that matches the CoreWeave bucket's location. Refer to [Regions and Availability Zones](https://docs.coreweave.com/docs/platform/regions) in the CoreWeave documentation. Click the **AI Object Storage** link for the reg ...
+- Replace `<coreweaveEndpoint>` with one of:
+  - `cwobject.com`: Primary HTTPS endpoint, TLS 1.3 required.
+  - `cwlota.com`: CoreWeave's [Local Object Transfer Accelerator (LOTA)](https://docs.coreweave.com/docs/products/storage/object-storage/concepts/lota) HTTP (not HTTPS) endpoint. LOTA is an intelligent proxy installed on every GPU Node in a CKS cluster to accelerate data transfer by providing an efficient local gateway to CoreWeave AI Object Storage on each node in the cluster for faster data tra ...
```
```suggestion
  The `cwobject.com` HTTPS endpoint is supported. TLS 1.3 is required. Contact [support](mailto:support@wandb.com) to express interest in other CoreWeave endpoints.
```

mdlinville committed:
Apply suggestions from code review

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -297,58 +325,101 @@ For more details, see the [AWS self-managed hosting guide]({{< relref "/guides/h
-{{% tab header="Team level" value="team" %}}
-If you're connecting to a cloud-native storage bucket in another cloud or to an S3-compatible storage bucket like [MinIO](https://github.com/minio/minio) for team-level BYOB in your [Dedicated cloud]({{< relref "/guides/hosting/hosting-options/dedicated_cloud.md" >}}) or [Self-managed]({{< relref "/guides/hosting/hosting-options/self-managed.md" >}}) instance, refer to [Cross-cloud or S3-compatib ...
+- CoreWeave uses availability zones for AI Object Storage. In the bucket address, the `region` parameter is mandatory and must be set to the name of the CoreWeave availability zone that matches the CoreWeave bucket's location. Refer to [Regions and Availability Zones](https://docs.coreweave.com/docs/platform/regions) in the CoreWeave documentation. Click the **AI Object Storage** link for the reg ...
+  The `cwobject.com` HTTPS endpoint is supported. TLS 1.3 is required. Contact [support](mailto:support@wandb.com) to express interest in other CoreWeave endpoints.
```
```suggestion

  The `cwobject.com` HTTPS endpoint is supported. TLS 1.3 is required. Contact [support](mailto:support@wandb.com) to express interest in other CoreWeave endpoints.
```

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -357,5 +428,80 @@ To make changes to the instance level bucket storage for your W&B Server:
+  - `ExposeHeaders` must include `ETag.
+  - W&B front-end domains must be included in the CORS policy's `AllowedOrigins`. The example CORS policies provided on this page include all domains using `*`.
+- **LOTA endpoint issues**
+  - LOTA endpoints use HTTP rather than HTTPS. Omit the `tls` parameter from the bucket's address, and use the `cw://` protocol specifier if possible.
```
```suggestion
  - Connecting to LOTA endpoints from W&B is not yet supported.  To express interest, [contact support](mailto:support@wandb.com).
```

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -357,5 +428,80 @@ To make changes to the instance level bucket storage for your W&B Server:
+- **LOTA endpoint issues**
+  - LOTA endpoints use HTTP rather than HTTPS. Omit the `tls` parameter from the bucket's address, and use the `cw://` protocol specifier if possible.
+  - Your W&B instance must be able to connect to `.cwlota.com`.
+  - Use LOTA only when your workloads are running on CoreWeave GPU compute. LOTA is optimized for large file transfers within CoreWeave infrastructure.
```
```suggestion
```

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -357,5 +428,80 @@ To make changes to the instance level bucket storage for your W&B Server:
+  - Use LOTA only when your workloads are running on CoreWeave GPU compute. LOTA is optimized for large file transfers within CoreWeave infrastructure.
+- **Region errors**
+  - CoreWeave uses availability zones for AI Object Storage.
+    - In the bucket address, set `region` to the name of the CoreWeave availability zone that matches the CoreWeave bucket's location.
```
```suggestion
    - In the bucket address, you can optionally set `region` to the name of the CoreWeave availability zone that matches the CoreWeave bucket's location. Because CoreWeave bucket names are globally unique, `region` is not required in the bucket address.
```

mdlinville committed:
Final read-through

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -297,58 +325,102 @@ For more details, see the [AWS self-managed hosting guide]({{< relref "/guides/h
 
-{{% tab header="Team level" value="team" %}}
-If you're connecting to a cloud-native storage bucket in another cloud or to an S3-compatible storage bucket like [MinIO](https://github.com/minio/minio) for team-level BYOB in your [Dedicated cloud]({{< relref "/guides/hosting/hosting-options/dedicated_cloud.md" >}}) or [Self-managed]({{< relref "/guides/hosting/hosting-options/self-managed.md" >}}) instance, refer to [Cross-cloud or S3-compatib ...
+- CoreWeave uses availability zones for AI Object Storage. In the bucket address, the `region` parameter is mandatory and must be set to the name of the CoreWeave availability zone that matches the CoreWeave bucket's location. Refer to [Regions and Availability Zones](https://docs.coreweave.com/docs/platform/regions) in the CoreWeave documentation. Click the **AI Object Storage** link for the reg ...
```
```suggestion
- CoreWeave uses availability zones for AI Object Storage. Refer to [Regions and Availability Zones](https://docs.coreweave.com/docs/platform/regions) in the CoreWeave documentation for details. Click the **AI Object Storage** link for the region to verify the ability zone to use. For example, in region `US-EAST-01`, AI Object Storage is available in `US-EAST-01A`. 
```

mdlinville commented on content/en/guides/hosting/data-security/secure-storage-connector.md:
```diff
@@ -297,58 +325,102 @@ For more details, see the [AWS self-managed hosting guide]({{< relref "/guides/h
+- CoreWeave uses availability zones for AI Object Storage. In the bucket address, the `region` parameter is mandatory and must be set to the name of the CoreWeave availability zone that matches the CoreWeave bucket's location. Refer to [Regions and Availability Zones](https://docs.coreweave.com/docs/platform/regions) in the CoreWeave documentation. Click the **AI Object Storage** link for the reg ...
+
+  The `cwobject.com` HTTPS endpoint is supported. TLS 1.3 is required. Contact [support](mailto:support@wandb.com) to express interest in other CoreWeave endpoints.
+  See the [CoreWeave documentation](https://docs.coreweave.com/docs/products/storage/object-storage/how-to/get-started-caios#3-create-a-bucket) for details.
```
```suggestion
```

mdlinville committed:
Apply suggestions from code review

mdlinville committed:
Fix min.io links
